soc/arm/st_stm32/stm32f0/Kconfig.defconfig.stm32f098xx

```diff
@@ -0,0 +1,14 @@
+# ST Microelectronics STM32F098XX MCU
+
+# Copyright (c) 2019 The Chromium OS Authors
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_STM32F098XX
+
+config SOC
+	def_string "stm32f098xx"
```

Comment:
Just make it

```
config SOC
	default "stm32f098xx"
```

and

```
config NUM_IRQS
	default 31
```

There's a "base" definition of these symbols outside the `Kconfig.defconfig` files (see the end of [the CONFIG_SOC reference page](https://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_SOC.html#cmdoption-arg-config-soc)). Those definitions already sets the type. Skipping the type in the `Kconfig.defconfig` files means the symbols become malformed if the base definition is removed. That makes left-over definitions easier to discover.

Reply:
Appreciate someone noticing `def_string` and `def_int` though. :)